models/cart-discount/src/cart-discount-draft/generator.ts

```diff
@@ -0,0 +1,42 @@
+import * as CartDiscountValueAbsolute from '@commercetools-test-data/cart-discount-value-absolute';
+import * as CartDiscountValueFixed from '@commercetools-test-data/cart-discount-value-fixed';
+import * as CartDiscountValueGiftLineItem from '@commercetools-test-data/cart-discount-value-gift-line-item';
+import * as CartDiscountValueRelative from '@commercetools-test-data/cart-discount-value-relative';
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { fake, Generator } from '@commercetools-test-data/core';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import { TCartDiscountDraft } from '../types';
+
+// https://docs.commercetools.com/api/projects/cartDiscounts#cartdiscountdraft
+
+// eslint-disable-next-line
+const [getValidFrom, _, getValidUntil] = createRelatedDates();
+
+const generator = Generator<TCartDiscountDraft>({
+  fields: {
+    key: fake((f) => f.lorem.slug(2)),
+    name: fake(() => LocalizedString.random()),
+    description: fake(() => LocalizedString.random()),
+    value: fake((f) =>
+      f.helpers.arrayElement([
+        CartDiscountValueAbsolute.CartDiscountValueAbsoluteDraft.random(),
+        CartDiscountValueFixed.CartDiscountValueFixedDraft.random(),
+        CartDiscountValueGiftLineItem.CartDiscountValueGiftLineItemDraft.random(),
+        CartDiscountValueRelative.CartDiscountValueRelativeDraft.random(),
+      ])
+    ),
+    cartPredicate: '1=1',
+    target: null,
+    sortOrder: fake((f) => String(Math.random())),
+    isActive: fake((f) => f.datatype.boolean()),
+    validFrom: fake(getValidFrom),
+    validUntil: fake(getValidUntil),
+    requiresDiscountCode: fake((f) => f.datatype.boolean()),
+    stackingMode: fake((f) =>
+      f.helpers.arrayElement(['Stacking', 'StopAfterThisDiscount'])
```

Comment:
This should be a constant ([example](https://github.com/commercetools/test-data/blob/main/models/channel/src/constants.ts)), then you could do something like f.helpers.arrayElement([Object.values(stackingMode)]) and we can later reference it as stackingMode.Stacking in a preset.

Reply:
Great strategy! Resolved in 0ff3350a6e7d3ff9f23e2f131992bd3cec4efc68